includes/class-wc-subscriptions-order.php

```diff
@@ -77,6 +77,22 @@ public static function init() {
 		add_filter( 'woocommerce_order_query_args', array( __CLASS__, 'map_order_query_args_for_subscriptions' ) );
 
 		add_filter( 'woocommerce_orders_table_query_clauses', [ __CLASS__, 'filter_orders_query_by_parent_orders' ], 10, 2 );
+
+		add_action( 'woocommerce_before_delete_order', [ __CLASS__, 'delete_order_update_order_related_subscriptions_last_order_date_created' ], 10, 2 );
+
+		$cache_manager = new WCS_Object_Data_Cache_Manager(
+			'subscription',
+			[
+				'parent_id',
+			]
+		);
+		$cache_manager->init();
+
+		add_action( 'wcs_update_post_meta_caches', [ __CLASS__, 'update_subscription_last_order_date_parent_id_changes' ], 10, 5 );
+
+		add_action( 'wcs_orders_add_relation', [ __CLASS__, 'add_relation_update_order_related_subscriptions_last_order_date_created' ], 10, 3 );
+
+		add_action( 'wcs_orders_delete_relation', [ __CLASS__, 'delete_relation_update_order_related_subscriptions_last_order_date_created' ], 10, 3 );
```

Comment:
This function name seems weird to me. Also it does not actually delete the order just update it

Reply:
@leonardola it deletes the order relation, not the order itself.